rgw/v2/tests/curl/test_rgw_using_curl.py

```diff
@@ -0,0 +1,152 @@
+            log.info(f"Bucket {bucket_name} created")
+
+            # create objects
+            objects_created_list = []
+            if config.test_ops.get("create_object", False):
+                # uploading data
+                log.info("s3 objects to create: %s" % config.objects_count)
+                for oc, size in list(config.mapped_sizes.items()):
+                    config.obj_size = size
+                    s3_object_name = utils.gen_s3_object_name(bucket_name, oc)
+                    log.info("s3 object name: %s" % s3_object_name)
+                    s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+                    log.info("s3 object path: %s" % s3_object_path)
+                    log.info("upload type: normal")
+                    curl_reusable.upload_object(
+                        curl_auth,
+                        bucket_name,
+                        s3_object_name,
+                        TEST_DATA_PATH,
+                        config,
+                        Transfer_Encoding=config.test_ops.get("Transfer_Encoding"),
+                    )
+                    objects_created_list.append(s3_object_name)
+                    if config.test_ops["download_object"] is True:
+                        curl_reusable.download_object(
+                            curl_auth,
+                            bucket_name,
+                            s3_object_name,
+                            TEST_DATA_PATH,
+                            s3_object_path,
+                        )
+                    if config.local_file_delete is True:
+                        log.info("deleting local file created after the upload")
+                        utils.exec_shell_cmd("rm -rf %s" % s3_object_path)
+            if config.test_ops.get("delete_bucket_object") is True:
+                for obj in objects_created_list:
+                    curl_reusable.delete_object(curl_auth, bucket_name, obj)
+                curl_reusable.delete_bucket(curl_auth, bucket_name)
+        if config.user_remove:
+            s3_reusable.remove_user(each_user)
+
+    # check for any crashes during the execution
+    crash_info = s3_reusable.check_for_crash()
+    if crash_info:
+        raise TestExecError("ceph daemon crash found!")
+
+
+if __name__ == "__main__":
+    test_info = AddTestInfo("test rgw operations using curl")
+
+    try:
+        project_dir = os.path.abspath(os.path.join(__file__, "../../.."))
+        test_data_dir = "test_data"
+        TEST_DATA_PATH = os.path.join(project_dir, test_data_dir)
+        log.info(f"TEST_DATA_PATH: {TEST_DATA_PATH}")
+        if not os.path.exists(TEST_DATA_PATH):
+            log.info("test data dir not exists, creating.. ")
+            os.makedirs(TEST_DATA_PATH)
+        parser = argparse.ArgumentParser(description="RGW S3 rgw operations using Curl")
+        parser.add_argument("-c", dest="config", help="RGW S3 operations using AWS")
```

Comment:
please correct the description accordingly

Reply:
@ckulal done, please review again!